core/src/main/java/com/homeaway/streamplatform/streamregistry/service/impl/StreamServiceImpl.java

```diff
@@ -0,0 +1,247 @@
+            }
+        }
+    }
+
+    private void upsertTopics(Stream stream, String vpc, boolean isNewStream) throws StreamCreationException, ClusterNotFoundException {
+        ClusterValue clusterValue = clusterService.getCluster(vpc, env, stream.getTags().getHint(), "producer");
+        Properties topicConfig = new Properties();
+        if (stream.getTopicConfig() != null) {
+            topicConfig.putAll(stream.getTopicConfig());
+        }
+
+        String bootstrapServer = clusterValue.getClusterProperties().get(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG);
+        String zkConnect = clusterValue.getClusterProperties().get(KafkaProducerConfig.ZOOKEEPER_QUORUM);
+        topicConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer);
+        topicConfig.put(KafkaProducerConfig.ZOOKEEPER_QUORUM, zkConnect);
+
+        kafkaManager.upsertTopics(Collections.singleton(stream.getName()), stream.getPartitions(), stream.getReplicationFactor(), topicConfig, isNewStream);
+        log.info("Topic {} created/updated at {}", stream.getName(), bootstrapServer);
+    }
+
+    private void applyDefaultHint(Stream stream) {
+        Tags tags = stream.getTags();
+
+        String hint = tags.getHint();
+
+        if (hint == null || hint.trim().matches("(?i:string)?")) {
+            hint = AbstractService.PRIMARY_HINT;
+        }
+
+        tags.setHint(hint);
+    }
+
+    @Override
+    public Stream getStream(String streamName) throws StreamNotFoundException {
+        log.info("Pulling stream information from global state-store for streamName={}", streamName);
+        Optional<AvroStream> streamValue = streamDao.getStream(streamName).getValue();
+
+        if (!streamValue.isPresent()) {
+            throw new StreamNotFoundException(String.format("Stream=%s not found.", streamName));
+        } else {
+            return AvroToJsonDTO.convertAvroToJson(streamValue.get());
+        }
+    }
+
+    @Override
+    public void deleteStream(String streamName) throws StreamNotFoundException{
+        Pair<AvroStreamKey, Optional<AvroStream>> keyValue = streamDao.getStream(streamName);
+
+        if (!keyValue.getValue().isPresent()) {
+            throw new StreamNotFoundException(String.format("Stream=%s not found.", streamName));
+        }
+        try {
+            streamDao.upsertStream(keyValue.getKey(), null);
+        } catch (Exception e) {
+            log.error("Error deleting stream", e);
+        }
+    }
+
+    @Override
+    public List<Stream> getAllStreams() {
+        List<Stream> streamList = new ArrayList<>();
+        log.info("Pulling stream information from local instance's state-store");
+        streamDao.getAllStreams().forEachRemaining(avroStream -> streamList.add(AvroToJsonDTO.convertAvroToJson(avroStream.value)));
+        return streamList;
+    }
+
+    @Override
+    public boolean validateSchemaCompatibility(Stream stream) throws SchemaValidationException {
```

Comment:
really need to fix this signature... either it should return boolean and not throw an exception
OR return void and throw an exception... not both return booland AND throw an exception.

Reply:
My preference... boolean (and not throw an exception)

Reply:
Fixed it.
gone with the throws Exception approach, and made it as an void statement, as we did lot of ground work for this approach in our previous PR